contrib/lkt/language/parser.py

```diff
@@ -2509,20 +2520,50 @@ def called_object_type():
     @langkit_property(public=True)
     def called_decl():
         """
-        Return the declaration that is called by this call expression.
+        Return the declaration that is called by this call expression, if there
+        is one that can be statically determined.
         """
-        refd_decl = Var(Entity.name.referenced_decl.then(
-            lambda sr: sr.result_ref
+
+        # We use this to resolve formals rather than just the type of the
+        # ``name`` of the callexpr, for two language features:
+
+        # * Generic functions: We need to statically resolve them in order to
+        # allow inference in the implicit instantiation.
+
+        # * Named parameters: Since function types don't contain parameter
+        # names, if we want to allow named parameters we need to statically
+        # resolve the target of the call when possible.
```

Comment:
I have the feeling we should allow named arguments even when the target of the call is not resolvable (for instance when dealing with property references, which don’t exist right now, but could be useful in the future). WDYT? (Ok to do that later of course)

Reply:
As discussed live: yes, let's do that (long term)